src/windows/ImagePickerProxy.js

```diff
@@ -0,0 +1,105 @@
+
+            var bitmapProperties = new Imaging.BitmapPropertySet();
+            var encoderId;
+            var fileType = inputFile.fileType.toUpperCase();
+            if (fileType === ".JPG" ||
+                fileType === ".JPEG") {
+                encoderId = Imaging.BitmapEncoder.jpegEncoderId;
+                var compression = Math.min(100, Math.max(0, options.quality));
+                compression = compression / 100.0;
+                bitmapProperties.insert("ImageQuality", new Imaging.BitmapTypedValue(compression, Windows.Foundation.PropertyType.single));
+
+            } else if (fileType === ".PNG") {
+                encoderId = Imaging.BitmapEncoder.pngEncoderId;
+            } else {
+                throw 'Invalid filetype'
+            }
+
+            return inputFile.openAsync(Windows.Storage.FileAccessMode.read).then(function (stream) {
+                inputStream = stream;
+                return tempFolder.createFileAsync("tmpImage", Windows.Storage.CreationCollisionOption.generateUniqueName);
+            }).then(function (file) {
+                outputFile = file;
+                return outputFile.openAsync(Windows.Storage.FileAccessMode.readWrite);
+            }).then(function (output) {
+                outputStream = output;
+                return Imaging.BitmapDecoder.createAsync(inputStream);
+            }.bind(this)).then(function (bitmapDecoder) {
+                decoder = bitmapDecoder;
+                return decoder.getPixelDataAsync();
+            }.bind(this)).then(function (pixelData) {
+                detachedPixelData = pixelData.detachPixelData();
+                return Imaging.BitmapEncoder.createAsync(encoderId, outputStream, bitmapProperties);
+            }.bind(this)).then(function (encoder) {
+                encoder.setPixelData(
+                    decoder.bitmapPixelFormat, decoder.bitmapAlphaMode,
+                    decoder.orientedPixelWidth, decoder.orientedPixelHeight, decoder.dpiX, decoder.dpiY,
+                    detachedPixelData);
+
+                var scaledSize = calculateScaledSize(
+                    { width: decoder.orientedPixelWidth, height: decoder.orientedPixelHeight },
+                    { width: options.width, height: options.height });
+
+                encoder.bitmapTransform.scaledWidth = scaledSize.width;
+                encoder.bitmapTransform.scaledHeight = scaledSize.height;
+
+                return encoder.flushAsync();
+            }.bind(this)).then(function () {
+                results.push('ms-appdata:///temp/' + outputFile.name);
+
+            }, function (err) {
+                error();
+            });
+        });
+            
+        return Promise.all(promises).then(function () {
+            success(results);
+        } , error);
+            
+    }, error);
+}
+
+ImagePicker = {
+    getPictures: function (success, error, params) {
+        var options = params[0];
+        getPictures(success, error, options);
```

Comment:
No need to define `options`, just use `params` inline (unless you need to check that `params` and `params[0]` exist).

```js
getPictures(success, error, params[0]);
```